crates/libcontainer/src/channel.rs

```diff
@@ -18,12 +18,13 @@ pub enum ChannelError {
     #[error("channel connection broken")]
     BrokenChannel,
 }
-
+#[derive(Clone)]
 pub struct Receiver<T> {
     receiver: RawFd,
     phantom: PhantomData<T>,
 }
 
+#[derive(Clone)]
 pub struct Sender<T> {
     sender: RawFd,
     phantom: PhantomData<T>,
```

Comment:
Given that this contains RawFd, is it fine to allow clone on this? Also, should we impose Clone on T as well (even if we are not storing that) ?

Reply:
> Clone RawFd

This is just a c_int, so this is safe to do. It is like pass the RawFd by value with copy. When across the fork/clone boundary, the file descriptor table is duplicated so existing fds works exactly the same.

> T for Clone

I don't think we have to impose `Clone` to T. In fact, it is possible that T does not have Clone trait. As long as T is serializable, we can use the channel to send and receive. So T can be serializable but not clone-able. The `PhantomData` does not store T as you pointed out and it is simply there to signal ownership of T.